core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -915,6 +915,33 @@ private fun getPendingRecording(
                     .build()
             videoCaptureUseCase.output.prepareRecording(context, mediaStoreOutput)
         }
+
+        is SaveLocation.Cache -> {
+            try {
+                // 1. Get the app's cache directory
+                val cacheDir = context.applicationContext.cacheDir
```

Comment:
Should this also try to use saveLocation.cacheDir before using the application's cacheDir?